rush-plugins/rush-amazon-s3-build-cache-plugin/src/RushAmazonS3BuildCachePlugin.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import { Import, JsonSchema } from '@rushstack/node-core-library';
```

Comment:
Be advised that, due to the auto-installers, this will be a separate runtime instance of `@rushstack/node-core-library` than the one that was loaded inside of Rush. This has performance implications.

Reply:
I thought about this before. Haven't found a better way now. 

Maybe we should declare `@rushstack/node-core-library` inside `pluginAutoinstaller`(enforced by rush), and the plugin itself should declare `@rushstack/node-core-library` as `peerDepenendency`. In this way, `@rushstack/node-core-library` should be the same runtime. but the drawback is too complicate to teach plugin developers.